### PR TITLE
Hotfix/close modal goes to activity

### PR DESCRIPTION
**Pull request by cliffmeyers**
# Description
Closing of the Run Modal was not navigating back to the previous route in many instances:
- Coming from the dashboard, after close it was navigating to /activity instead of dashboard
- Coming from /branches, closing was navigating to /activity
- Deep-linking directly to Run Details /pipeline then closing was navigating back to /activity correctly
- Deep-linking to Run Details with no tab specified then closing was navigating back to /activity correctly
- Verified the above scenarios. Wrote a few unit tests for the buggy code in question

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by cliffmeyers** [comment]
Pending local ATH run...

**Comment by cliffmeyers** [comment]
```
2017-01-10 20:53:40,791 [Thread-3] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - OK. 1027  total assertions passed. (9m 24s)
2017-01-10 20:53:40,861 [Thread-3] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - [20:53:40] Finished 'default' after 9.4 min
**** NightwatchRunner - No failures.
```

File: blueocean-core-js/src/js/services/LocationService.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+ * Stores the previous and current location pathnames.
  */
 export default class LocationService {
     @observable current;
     @observable previous;
 
     @action setCurrent(newLocation) {
         if (newLocation.action !== 'REPLACE') {
-            this.current = this.previous;
+            this.previous = this.current;
```
Comment (sophistifunk) [inline blueocean-core-js/src/js/services/LocationService.js]: 👍

**Comment by sophistifunk** [comment]
🐝

**Comment by cliffmeyers** [comment]
ATH passed again.

```
017-01-10 21:18:52,896 [Thread-3] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - OK. 1094  total assertions passed. (10m 37s)
2017-01-10 21:18:52,961 [Thread-3] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - [21:18:52] Finished 'default' after 11 min
**** NightwatchRunner - No failures.
```

**Pull request merged**